### PR TITLE
pilot: clusterregistry cleanup and fix

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. remove RemoteCluster.client

1. handle DeletedFinalStateUnknown type to prevent type cast panic in kubernetes service registry.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller.go [hunk 1]
```diff
@@ -541,7 +541,19 @@ func (c *Controller) GetIstioServiceAccounts(hostname model.Hostname, ports []in
 // AppendServiceHandler implements a service catalog operation
 func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
-		svc := *obj.(*v1.Service)
+		svc, ok := obj.(*v1.Service)
+		if !ok {
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				log.Errorf("Couldn't get object from tombstone %#v", obj)
+				return nil
```
Comment (rshriram) [inline pilot/pkg/serviceregistry/kube/controller.go]: Do we need to log this? Too much noise ?

**Comment by john-a-joyce** [comment]
I am working on a PR that converts pilot to use the common secret library.  It is still WIP but it will remove the clusterregistry files in pilot entirely.   See https://github.com/istio/istio/pull/8479
  It is possible some of these changes should be applied to the common library that was added in:  https://github.com/istio/istio/pull/7574.   I haven't analyzed the code yet to check for that.

File: pilot/pkg/serviceregistry/kube/controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller.go]: @rshriram Not too many such logs. And kubernetes also log such message. This only can be produced when a delete event lost.  If you feel strongly, I can delete this log.

**Pull request merged**